src/main/java/org/opensearch/timeseries/rest/handler/AbstractTimeSeriesActionHandler.java

```diff
@@ -663,28 +616,104 @@ private void onGetConfigResponse(GetResponse response, boolean indexingDryRun, S
             } else {
                 if (!ParseUtils.listEqualsWithoutConsideringOrder(existingConfig.getCategoryFields(), config.getCategoryFields())
                     || !Objects.equals(existingConfig.getCustomResultIndexOrAlias(), config.getCustomResultIndexOrAlias())) {
                     breakingUIChange = true;
                 }
             }
-            handleFlattenResultIndexMappingUpdate(existingConfig, listener);
+            ActionListener<Void> confirmBatchRunningListener;
 
-            ActionListener<Void> confirmBatchRunningListener = ActionListener
-                .wrap(
-                    r -> searchConfigInputIndices(id, indexingDryRun, listener),
-                    // can't update config if there is task running
-                    listener::onFailure
-                );
+            // when unselecting flatten result index, need to clean up alias and ingest pipeline resources
+            if (existingConfig.getFlattenResultIndexMapping()
+                && !config.getFlattenResultIndexMapping()
+                && existingConfig.getCustomResultIndexOrAlias() != null) {
+                confirmBatchRunningListener = ActionListener
+                    .wrap(
+                        r -> getFlattenResultAliasIndex(existingConfig, listener, id, indexingDryRun),
+                        // can't update config if there is task running
+                        listener::onFailure
+                    );
 
+            } else {
+                confirmBatchRunningListener = ActionListener
+                    .wrap(
+                        r -> searchConfigInputIndices(id, indexingDryRun, listener),
+                        // can't update config if there is task running
+                        listener::onFailure
+                    );
+            }
             handler.confirmBatchRunning(id, batchTasks, confirmBatchRunningListener);
         } catch (Exception e) {
             String message = "Failed to parse config " + id;
             logger.error(message, e);
             listener.onFailure(new OpenSearchStatusException(message, RestStatus.INTERNAL_SERVER_ERROR));
         }
+    }
+
+    private void getFlattenResultAliasIndex(Config existingConfig, ActionListener<T> listener, String id, boolean indexingDryRun) {
+        String flattenResultIndexAlias = timeSeriesIndices
+            .getFlattenedResultIndexAlias(existingConfig.getCustomResultIndexOrAlias(), existingConfig.getId());
+        GetAliasesRequest getAliasesRequest = new GetAliasesRequest(flattenResultIndexAlias);
+        client.admin().indices().getAliases(getAliasesRequest, ActionListener.wrap(getAliasesResponse -> {
+            Set<String> indices = getAliasesResponse.getAliases().keySet();
+            if (indices.isEmpty()) {
+                return;
+            }
+            String indexName = indices.iterator().next();
+            deleteAlias(indexName, flattenResultIndexAlias, existingConfig, listener, id, indexingDryRun);
```

Comment:
Why do you want to delete alias? Unselecting flatten option causes data deletion does not seem to be a good user experience.